packages/blade/src/components/Amount/Amount.tsx

```diff
@@ -0,0 +1,264 @@
+  const props: ColorProps = {
+    amountValueColor: 'surface.text.normal.lowContrast',
+    affixColor: 'surface.text.muted.lowContrast',
+  };
+  if (!intent && intent !== 'neutral') return props;
+  props.amountValueColor = `feedback.text.${intent}.lowContrast`;
+  props.affixColor = `feedback.text.${intent}.lowContrast`;
+  return props;
+};
+
+interface AmountValue extends Omit<AmountProps, 'value'> {
+  affixColor: BaseTextProps['color'];
+  amountValueColor: BaseTextProps['color'];
+  value: string;
+  size: Exclude<AmountProps['size'], undefined>;
+}
+
+const AmountValue = ({
+  value,
+  size,
+  amountValueColor,
+  isAffixSubtle,
+  suffix,
+  affixColor,
+}: AmountValue): ReactElement => {
+  const affixFontWeight = isAffixSubtle ? 'regular' : 'bold';
+  const affixFontSize = isAffixSubtle ? affixFontSizes[size] : amountFontSizes[size];
+  const valueForWeight = size.includes('bold') || size.startsWith('title') ? 'bold' : 'regular';
+  if (suffix === 'decimals' && isAffixSubtle) {
+    const integer = value.split('.')[0];
+    const decimal = value.split('.')[1];
+
+    // Native does not support alignItems of Text inside a div, insted we need to wrap is in a Text
+    const AmountWrapper = getPlatformType() === 'react-native' ? BaseText : React.Fragment;
+
+    return (
+      <AmountWrapper>
+        <BaseText
+          fontSize={amountFontSizes[size]}
+          fontWeight={valueForWeight}
+          color={amountValueColor}
+        >
+          {integer}.
+        </BaseText>
+        <BaseText
+          marginLeft="spacing.1"
+          fontWeight={affixFontWeight}
+          fontSize={affixFontSize}
+          color={affixColor}
+        >
+          {decimal || '00'}
+        </BaseText>
+      </AmountWrapper>
+    );
+  }
+  return (
+    <BaseText fontSize={amountFontSizes[size]} fontWeight={valueForWeight} color={amountValueColor}>
+      {value}
+    </BaseText>
+  );
+};
+
+// This function rounds a number to a specified number of decimal places
+// and floors the result.
+const getFlooredFixed = (value: number, decimalPlaces: number): number => {
+  const factor = 10 ** decimalPlaces;
+  const roundedValue = Math.floor(value * factor) / factor;
+  return Number(roundedValue.toFixed(decimalPlaces));
+};
+
+const addCommas = (amountValue: number, currency: Currency): string => {
+  const locale = currencyLocaleMapping[currency];
+  return amountValue.toLocaleString(locale);
+};
+
+const getHumanizedAmount = (amountValue: number, currency: Currency): string => {
+  const abbreviations = currencyAbbreviationsMapping[currency];
+
+  const abbreviation = abbreviations.find((abbr) => amountValue >= abbr.value);
+  if (abbreviation) {
+    amountValue = amountValue / abbreviation.value;
+    const formattedAmountValue = getFlooredFixed(amountValue, 2);
+    return addCommas(formattedAmountValue, currency) + abbreviation.symbol;
+  } else {
+    return amountValue.toString();
+  }
+};
```

Comment:
We should add some unit tests for these utils

Reply:
While taking snapshots, we are conversing this.

Reply:
Yup, but having few tests for these utils will be helpful for future maintenance. Just 2 - 3 tests per utils should do to cover possible scenarios.